Add an isObject function and use it.

jloveridge opened the pull request:
I attempted to use sift in a project where we use a customized REPL and encountered an issue where things didn't filter properly. All of the examples were returning an empty array instead of the expected values. The problem turned out to be the way the statement was being checked to see if it was an Object. I snagged the `isObject` code from `lodash` and used that for the check instead and the issue goes away. The issue can be observed for yourself by creating any custom REPL environment. The following would do:

```javascript
require('repl').start({prompt: 'custom> '})
```

Then using it:
```
$ node custom-repl.js
custom> var sift = require('sift');
undefined
custom> sift({ $in: ['hello','world'] }, ['hello','sifted','array!']);
[]
custom> 
```

crcn commented:
Thanks for the fix! Looks like there are just a few linting issues. Mind fixing those and re-submitting a PR?

jloveridge commented:
Okay, I have fixed up the lint errors and changed the methodology for testing of the statement. This one seems to pass all of the tests locally and I see that it just succeeded on Travis build. Had to add a second check on the toString of the constructor to allow REPL implementations to work.

jloveridge commented:
With my original change it was causing a bunch of failures in the tests due to arrays, dates, and the ObjectID failing to use the `eq` test. Resolving the issue for dates and arrays was simple enough but custom objects proved more problematic. The `ObjectID` tests failed inside of the `toHexString` as `this` wasn't an `ObjectID` but rather `{test: function...}`. (I still don't know how it ended up calling that function.)

I have tested the latest both inside the standard node shell and within a custom REPL and everything now seems to work.

crcn commented:
Awesome. Thanks for doing all this - looks really good.

jloveridge commented:
I just thought about this but I wonder if it would be better to just do:

```javascript
if (statement && statement.constructor.toString() === Object.constructor.toString())
```
Wish I had thought some time ago.
